controllers/toolchaincluster_resources/toolchaincluster_resources_controller.go

```diff
@@ -0,0 +1,68 @@
+package toolchaincluster_resources
+
+import (
+	"context"
+	"embed"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	applycl "github.com/codeready-toolchain/toolchain-common/pkg/client"
+	"github.com/codeready-toolchain/toolchain-common/pkg/template"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.ToolchainCluster{}).
```

Comment:
The manifests should be applied in the cluster even before the very first ToolchainCluster CR is created in the namespace. The creation of the ToolchainCluster CR relies on the existence of these resources.
I would watch ServiceAccounts as the main driver for this controller. We know that there is gonna be always at least one SA in the namespace, so we know that the controller will be always triggered when the operator is started for the first time.

Watch for ToolchainCluster CRs will be added later as part of this story [KUBESAW-44](https://issues.redhat.com/browse/KUBESAW-44)

Reply:
Yup, I've updated the code to watch for SA and all the other types from the templates. Thanks for your pointers!